rasterio/rio/warp.py

```diff
@@ -260,19 +258,26 @@ def warp(ctx, files, output, driver, like, dst_crs, dimensions, src_bounds,
                 'height': dst_height
             })
 
+            # Adjust block size if necessary.
+            if ('blockxsize' in out_kwargs and
+                    dst_width < out_kwargs['blockxsize']):
+                del out_kwargs['blockxsize']
+            if ('blockysize' in out_kwargs and
+                    dst_height < out_kwargs['blockysize']):
+                del out_kwargs['blockysize']
```

Comment:
Getting `out_kwargs` from `src.profile` instead of `src.meta` (which means the output gets the input's layout by default) triggered this. Good thing we were testing with little 10x10 images :)